career_development/ladder.md

```diff
@@ -0,0 +1,126 @@
+	
+#### Hard skills
+- [ ] Can use git and linters.
+- [ ] Can write code in at least one language.
+
+### Level 2
+#### Simple 	
+- [ ] Creates simple method / functions that most static code analysers would accept.
+- [ ] Makes clear prs
+#### Test-driven
+- [ ] Codes unit tests before code, writes acceptance tests.
+#### Iterative
+- [ ] Delivers working code with each commit.
+#### Communication	
+- [ ] Hosts retros.
+- [ ] Can compose a PR and provide feedback on them.
+#### Respectful teammate
+- [ ] Shows respect for client goals
+- [ ] Encourages team to stay on a sustainable pace.
+#### Hard skills
+- [ ] Understands what differentiates their favourite language with others. Knows what the language is useful for and when it is not the best option.
+- [ ] Understand where to place logic in their favourite framework.
+
+### Level 3
+#### Simple 
+- [ ]
+#### Test-driven
+- [ ] Can start a project from scratch and setup testing frameworks.
+- [ ] Has an opinion about which frameworks to use.
+- [ ] Can mock http requests and understands where these tests will potentially fail.
+- [ ] Can integrate CI.	
+#### Iterative	
+- [ ] Works with red-green-refactor.
+- [ ] Uses time management techniques
+#### Communication	
+- [ ] Is proficient with pair programming.
+- [ ] Can write user stories.
+- [ ] Can introduce a new member to the project. 
+#### Respectful teammate	
+- [ ] Actively participates in feedback for team members.
+#### Hard skills
+- [ ] Can write code in at least two technical areas (frontend, backend, digital infrastructure). 
+
+### Level 4
+#### Simple 
+- [ ] Does not over optimize. Understands when code is "good-enough".
+#### Test-driven	
+- [ ] Makes use of stubbing and mocking.
+- [ ] Understands tradeoffs of hitting the db. 
+#### Iterative	
+- [ ] Well-composed commit narratives in commit flows. 
+#### Communication	
+- [ ] Can help PM cultivate an Asana board.
+- [ ] Understands kanban.
+- [ ] Can teach juniors to pair program.
+#### Respectful teammate
+- [ ] Communicate with PMs when goals are unattainable.
+- [ ] Understands the difficulties that the team is going through and helps how they can.	
+#### Hard skills
+- [ ] Demonstrates ability to succeed in a wide range of complex situations across multiple axes: e.g. scale, uncertainty, interconnectedness.
+
+### Level 5
+#### Simple 	
+- [ ] Extracts complex logic into small easy-to-follow classes/components.
+#### Test-driven
+- [ ]	
+#### Iterative	
+- [ ] Can guide an IPM and team reflection.
+#### Communication	
+- [ ] Aligns team on expectations.
+- [ ] Identifies miscommunications and solves them before they become a problem.
+#### Respectful teammate
+- [ ] Help dissolve tension when arises.
+- [ ] Does not take "all" responsibilities. They let their teammates take responsibilities they care about.	
+#### Hard skills
+- [ ] Breaks down complex projects into simple systems that can be effectively built and maintained by less experienced engineers.
```

Comment:
I think this one is really good - seems very in place with the level and everything - but maybe we should decide on a single word to use?' I'm thinking about the "engineers"-part where in other places we call it "developer" I don't think the whole architect/engineer/developer discussion is very beneficial and since I assume that we're trying to include everyone who writes code it might be enough to simple use "team member"? I mean, we _are_ one big team(a bit mushy, I know 😉) and this is a "dev career ladder".

Reply:
Good point :+1: